src/pybind/mgr/cephadm/module.py

```diff
@@ -467,6 +467,16 @@ def _check_for_strays(self):
                 missing_names = []
                 for s in daemons:
                     name = '%s.%s' % (s.get('type'), s.get('id'))
+                    if s.get('type') == 'rbd-mirror':
+                        try:
```

Comment:
The try clause probably is not needed:
It seems that you want to use it to detect if there is no information about the 'rbd-mirror' service. But this is impossible to happen:
if you pass the condition s.get('type') == 'rbd-mirror', you have the information you need for sure. ( this means that " rbd-mirror" is one of the daemons runnnig in one server, and therefore the metadata information, and over all the "id" is going to be present in the metadata section. If not is present this information (the id of the service), then what we have is a bug in the MgrModule.getMetadata method.) 

I thing you can do directly:
```
if s.get('type') == 'rbd-mirror':
    name = '%s.%s' % ('rbd-mirror', self.get_metadata('rbd-mirror', s.get('id'))['id'] )
```